internal/cache/file/downloader/job_test.go

```diff
@@ -571,9 +574,7 @@ func (dt *downloaderTest) Test_Download_Concurrent() {
 	objectName := "path/in/gcs/foo.txt"
 	objectSize := 25 * util.MiB
 	objectContent := testutil.GenerateRandomBytes(objectSize)
-	var callbackExecutionCount atomic.Int32
-	removeCallback := func() { callbackExecutionCount.Add(1) }
-	dt.initJobTest(objectName, objectContent, DefaultSequentialReadSizeMb, uint64(objectSize*2), removeCallback)
+	dt.initJobTest(objectName, objectContent, DefaultSequentialReadSizeMb, uint64(objectSize*2), nil)
```

Comment:
I think we shouldn't remove the `removeCallback` test.  Any particular reason for removing this ?

Reply:
There is no way to wait for the downloadObjectAsync to be executed completed. We are either waiting for the CRC check to be completed or the notification. These 2 doesn't mean the entire function is executed. So the defer is not getting called before our Assert statement. 

We should add waits to do that validation. I dont prefer adding waits since the time keep on changing everytime logic is changed and increases test time execution.